api/v1beta1/network_disruption.go

```diff
@@ -458,3 +454,46 @@ func (h NetworkDisruptionHostSpec) Validate() error {
 
 	return nil
 }
+
+func (s NetworkDisruptionServiceSpec) ExtractAffectedPortsInServicePorts(k8sService *v1.Service) ([]v1.ServicePort, []NetworkDisruptionServicePortSpec) {
+	if len(s.Ports) == 0 {
+		return k8sService.Spec.Ports, nil
+	}
+
+	servicePortsDic := map[string]v1.ServicePort{}
+	goodPorts, notFoundPorts := []v1.ServicePort{}, []NetworkDisruptionServicePortSpec{}
+
+	// Convert service ports from found k8s service to a dictionary in order to facilitate the filtering of the ports
+	for _, port := range k8sService.Spec.Ports {
+		servicePortsDic[strconv.Itoa(int(port.Port))] = port
+		if port.Name != "" {
+			servicePortsDic[port.Name] = port
+		}
```

Comment:
if someone is weirdly defining a name of the port as a number into the k8s service, you might end up with an issue.

Maybe use a prefix to guarantee you know what you are looking for (either a number or a name, that can also be a number)

```suggestion
		servicePortsDic[fmt.Sprintf("port-%d", port.Port)] = port
		if port.Name != "" {
			servicePortsDic[fmt.Sprintf("name-%s", port.Name)] = port
		}
```